django/db/backends/utils.py

```diff
@@ -65,14 +66,27 @@ def callproc(self, procname, params=None, kparams=None):
                 return self.cursor.callproc(procname, params, kparams)
 
     def execute(self, sql, params=None):
+        return self._execute_with_wrappers(sql, params, False)
+
+    def executemany(self, sql, param_list):
+        return self._execute_with_wrappers(sql, param_list, True)
+
+    def _execute_with_wrappers(self, sql, params, many):
+        executor = self._execute if not many else self._executemany
+        context = dict(connection=self.db, cursor=self)
+        for wrapper in reversed(self.db.get_execute_wrappers(many)):
```

Comment:
the `reversed` call isn't free, it's slightly more optimal to put the wrappers in the list in the way you want to iterate them

Reply:
The differences are infinitesimal (on my system, each `reversed()` call takes ~0.2 microseconds). I think it's more readable and debuggable this way -- when the wrappers are held in the order installed.

Note also that `append()` is significantly faster than `insert(0)` -- we expect quite short lists here, so it doesn't really matter, but in general, unless you read the list many times, append+reversed is the way to go:
```python
In [3]: %timeit x=[]; [x.append(i) for i in range(30)]; reversed(x)
4.41 µs ± 203 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [4]: %timeit x=[]; [x.insert(0,i) for i in range(30)]; 
6.28 µs ± 31.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [5]: %timeit x=[]; [x.append(i) for i in range(30)]; reversed(x); reversed(x); reversed(x)
4.77 µs ± 52.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
It seems that for a list of length 30, you'd need to read it more than 20 times for `insert(0)` to pay off.

Reply:
fair enough, I guess I'm just a bit too obsessed with optimization :)

Reply:
btw `reversed(x)` doesn't actually iterate the whole list in reverse in python 3, you just get a `list_reverseiterator`...

```
In [1]: reversed([1])
Out[1]: <list_reverseiterator at 0x105098b70>
```

:)